store/queries/candidate_posts.sql

```diff
@@ -21,8 +21,15 @@ FROM
 WHERE
       cp.is_hidden = false
   AND ca.status = 'approved'
-  AND (@require_tags::TEXT[] = '{}' OR @require_tags::TEXT[] <@ cp.tags)
-  AND (@exclude_tags::TEXT[] = '{}' OR NOT (@exclude_tags::TEXT[] && cp.tags))
+  AND (
+    (@require_tags::TEXT[] = '{}' OR @require_tags::TEXT[] <@ cp.tags) OR
+    (@require_hashtags::TEXT[] = '{}' OR @require_hashtags::TEXT[] <@ cp.hashtags)
+  )
+  AND (
+    (@exclude_tags::TEXT[] = '{}' OR NOT (@exclude_tags::TEXT[] && cp.tags)) AND
+    (@exclude_hashtags::TEXT[] = '{}' OR NOT (@exclude_hashtags::TEXT[] && cp.hashtags))
+  )
+  AND (NOT @must_have_media::BOOLEAN OR COALESCE(cp.has_media, true))
```

Comment:
I'd be tempted to make this a nullable boolean so we can make list posts w/o media rather than just posts w/ media.